# Análise de Custos de Saúde

## Problema / Situação


- Considerando que os custos de saúde têm aumentado muito e baixa qualidade de vida afetando a produtividade, as empresas buscam soluções de qualidade de vida que possam reduzir esses gastos em saúde(o plano de saúde é um benefício onde o reajuste anual é  baseado na utilização dos colaboradores). Com cada vez mais empresas usando a Wellbe é necessário garantir agilidade e acurácia no processo de população do banco de dados.


## Desafio
Realizar uma implementação completa dos dados, desde extração, transformação e carregamento dos dados para a construção de um dashboard.

## Extração dos Dados

- Importando a biblioteca pandas

In [1]:
import pandas as pd

- Lendo o arquivo base

In [2]:
df = pd.read_excel('Dados_Desafio.xlsx')

- Olhando as 15 primeiras linhas

In [3]:
df.head(15)

Unnamed: 0            Unnamed: 1     Unnamed: 2   Unnamed: 3  \
0                     NaN                   NaN            NaN          NaN   
1                     NaN                   NaN            NaN          NaN   
2                     NaN                   NaN            NaN          NaN   
3                     NaN                   NaN            NaN          NaN   
4                     NaN                   NaN            NaN          NaN   
5                     NaN                   NaN            NaN          NaN   
6                     NaN                   NaN            NaN          NaN   
7                     NaN                   NaN            NaN          NaN   
8   Listagem de Atestados                   NaN            NaN          NaN   
9               Empresa:                    NaN            NaN          NaN   
10                    NaN                   NaN            NaN          NaN   
11                    NaN                   NaN            NaN          NaN   
12                 Código  Custo do afastamento  Identificação  Funcionário   
13                1036743                  20,4            NaN  Anonimo 101   
14                1036742                    23            NaN    Anonimo 1   

                   Unnamed: 4        Unnamed: 5             Unnamed: 6  \
0                         NaN               NaN                    NaN   
1                         NaN               NaN                    NaN   
2                         NaN               NaN                    NaN   
3                         NaN               NaN                    NaN   
4                         NaN               NaN                    NaN   
5                         NaN               NaN                    NaN   
6                         NaN               NaN                    NaN   
7                         NaN               NaN                    NaN   
8                         NaN               NaN                    NaN   
9                         NaN               NaN                    NaN   
10                        NaN               NaN                    NaN   
11                        NaN               NaN                    NaN   
12               Departamento  Data do Atestado          Especialidade   
13                    Gerente        29/05/2019                 Exames   
14  ASSISTENTE DE IMPLANTACAO        23/05/2019  Neurologia pediátrica   

                 Unnamed: 7 Unnamed: 8  
0                       NaN        NaN  
1                       NaN        NaN  
2                       NaN        NaN  
3                       NaN        NaN  
4                       NaN        NaN  
5                       NaN        NaN  
6                       NaN        NaN  
7                       NaN        NaN  
8                       NaN        NaN  
9                       NaN        NaN  
10                      NaN        NaN  
11                      NaN        NaN  
12                   Motivo      Líder  
13                   Exames        Sim  
14  Acompanhamento familiar        NaN

## Tratamendo dos dados

- Podemos ver que nossos dados começam apenas na linha 12, as linhas acima não tem informações relevantes, vamo retirar as 11 primeiras linhas do nosso dataframe:

In [4]:
df2 = df.drop(df.index[:12])

df2

Unnamed: 0            Unnamed: 1     Unnamed: 2   Unnamed: 3  \
12      Código  Custo do afastamento  Identificação  Funcionário   
13     1036743                  20,4            NaN  Anonimo 101   
14     1036742                    23            NaN    Anonimo 1   
15     1036741                  22,8            NaN    Anonimo 1   
16     1036740                    –-            NaN    Anonimo 1   
..         ...                   ...            ...          ...   
99     1037659                 95.24            NaN   Anonimo 21   
100    1027619                 95.24            NaN   Anonimo 21   
101    1036380                   NaN            NaN   Anonimo 22   
102    1032392                 99.16            NaN   Anonimo 22   
103    1027618                 99.16            NaN   Anonimo 22   

                       Unnamed: 4        Unnamed: 5             Unnamed: 6  \
12                   Departamento  Data do Atestado          Especialidade   
13                        Gerente        29/05/2019                 Exames   
14      ASSISTENTE DE IMPLANTACAO        23/05/2019  Neurologia pediátrica   
15      ASSISTENTE DE IMPLANTACAO        17/05/2019                    NaN   
16      ASSISTENTE DE IMPLANTACAO        16/05/2019                 Exames   
..                            ...               ...                    ...   
99           ANALISTA SINISTROS I        04/06/2019           Oftalmologia   
100          ANALISTA SINISTROS I        13/03/2019              Pediatria   
101                    AUDITOR II        31/05/2019          Cardiologista   
102  ANALISTA ESTUDOS E COTACAO I        23/04/2019           Odontologia    
103  ANALISTA ESTUDOS E COTACAO I        12/03/2019           Odontologia    

                  Unnamed: 7 Unnamed: 8  
12                    Motivo      Líder  
13                    Exames        Sim  
14   Acompanhamento familiar        NaN  
15           Consulta médica        NaN  
16                    Exames        NaN  
..                       ...        ...  
99   Acompanhamento familiar        NaN  
100  Acompanhamento familiar        NaN  
101                   Exames        NaN  
102    Consulta odontológica        NaN  
103    Consulta odontológica        NaN  

[92 rows x 9 columns]

- Agora temos a primeira linha do dataframe com os nomes das colunas, mas queremos que elas sejam o cabeçalho do nosso dataframe, iremos também remover a coluna "Identificação", já que ela só tem valores nulos.

In [5]:
df2.columns = df2.iloc[0]  # Define os nomes das colunas com base na primeira linha

df2 = df2[1:] #remove a primeira linha que agora é o nome das colunas 

df2 = df2.drop('Identificação', axis=1) # Remove a coluna de identificação

df2.to_csv('dados_tratados.csv', index=False) # Salva o arquivo em csv


df2.head(15)

12   Código Custo do afastamento  Funcionário  \
13  1036743                 20,4  Anonimo 101   
14  1036742                   23    Anonimo 1   
15  1036741                 22,8    Anonimo 1   
16  1036740                   –-    Anonimo 1   
17  1036739                   22    Anonimo 1   
18  1033172                  205    Anonimo 1   
19  1032228                  123    Anonimo 1   
20  1030902                  205    Anonimo 1   
21  1036621                  321    Anonimo 2   
22  1036749                 99.8  Anonimo 114   
23  1036762                  331    Anonimo 3   
24  1027641                  123    Anonimo 4   
25  1036760                92.36    Anonimo 5   
26  1037656                94.61  Anonimo 109   
27  1036733                94.61    Anonimo 9   

12                     Departamento Data do Atestado          Especialidade  \
13                          Gerente       29/05/2019                 Exames   
14        ASSISTENTE DE IMPLANTACAO       23/05/2019  Neurologia pediátrica   
15        ASSISTENTE DE IMPLANTACAO       17/05/2019                    NaN   
16        ASSISTENTE DE IMPLANTACAO       16/05/2019                 Exames   
17        ASSISTENTE DE IMPLANTACAO       06/05/2019                    NaN   
18        ASSISTENTE DE IMPLANTACAO       24/04/2019              Pediatria   
19        ASSISTENTE DE IMPLANTACAO       04/04/2019  Neurologia pediátrica   
20        ASSISTENTE DE IMPLANTACAO       07/03/2019           Odontologia    
21               TECNICO SEGUROS IV       21/05/2019                    NaN   
22                          Gerente       02/05/2019                    NaN   
23               TECNICO SEGUROS II       09/05/2019                    NaN   
24  ANALISTA CONTROLE OPERACIONAL I       13/03/2019           Dermatologia   
25        ASSISTENTE OPERACIONAL II       16/05/2019             Neurologia   
26                          Gerente       18/06/2019              Nutrólogo   
27             TECNICO SEGUROS VG I       27/05/2019                    NaN   

12                   Motivo Líder  
13                   Exames   Sim  
14  Acompanhamento familiar   NaN  
15          Consulta médica   NaN  
16                   Exames   NaN  
17          Consulta médica   NaN  
18  Acompanhamento familiar   NaN  
19  Acompanhamento familiar   NaN  
20    Consulta odontológica   NaN  
21                      NaN   NaN  
22          Consulta médica   Sim  
23                      NaN   NaN  
24          Consulta médica   NaN  
25          Consulta médica   NaN  
26          Consulta médica   Sim  
27                      NaN   NaN

- Agora que ja temos os nossos dados arrumados, vamos para uma análise mais profunda

- Vamos ler os dados salvos em csv

In [6]:
df_novo = pd.read_csv('dados_tratados.csv')

df_novo

Código Custo do afastamento  Funcionário                  Departamento  \
0   1036743                 20,4  Anonimo 101                       Gerente   
1   1036742                   23    Anonimo 1     ASSISTENTE DE IMPLANTACAO   
2   1036741                 22,8    Anonimo 1     ASSISTENTE DE IMPLANTACAO   
3   1036740                   –-    Anonimo 1     ASSISTENTE DE IMPLANTACAO   
4   1036739                   22    Anonimo 1     ASSISTENTE DE IMPLANTACAO   
..      ...                  ...          ...                           ...   
86  1037659                95.24   Anonimo 21          ANALISTA SINISTROS I   
87  1027619                95.24   Anonimo 21          ANALISTA SINISTROS I   
88  1036380                  NaN   Anonimo 22                    AUDITOR II   
89  1032392                99.16   Anonimo 22  ANALISTA ESTUDOS E COTACAO I   
90  1027618                99.16   Anonimo 22  ANALISTA ESTUDOS E COTACAO I   

   Data do Atestado          Especialidade                   Motivo Líder  
0        29/05/2019                 Exames                   Exames   Sim  
1        23/05/2019  Neurologia pediátrica  Acompanhamento familiar   NaN  
2        17/05/2019                    NaN          Consulta médica   NaN  
3        16/05/2019                 Exames                   Exames   NaN  
4        06/05/2019                    NaN          Consulta médica   NaN  
..              ...                    ...                      ...   ...  
86       04/06/2019           Oftalmologia  Acompanhamento familiar   NaN  
87       13/03/2019              Pediatria  Acompanhamento familiar   NaN  
88       31/05/2019          Cardiologista                   Exames   NaN  
89       23/04/2019           Odontologia     Consulta odontológica   NaN  
90       12/03/2019           Odontologia     Consulta odontológica   NaN  

[91 rows x 8 columns]

- Depois de ler o novo csv, vamos usar a função "info" para saber os tipos de dados das colunas

In [7]:
df_novo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91 entries, 0 to 90
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Código                91 non-null     int64 
 1   Custo do afastamento  90 non-null     object
 2   Funcionário           91 non-null     object
 3   Departamento          89 non-null     object
 4   Data do Atestado      90 non-null     object
 5   Especialidade         75 non-null     object
 6   Motivo                85 non-null     object
 7   Líder                 8 non-null      object
dtypes: int64(1), object(7)
memory usage: 5.8+ KB


- Podemos ver que todas as colunas estão do tipo "object", vamos mudar a coluna "Data do Atestado" para o tipo datetime.

In [8]:
import numpy as np

# Removendo valores nulos

df_novo['Custo do afastamento'] = df_novo['Custo do afastamento'].replace("–-", np.nan)



# Transformando a coluna de data em datetime
df_novo['Data do Atestado'] = pd.to_datetime(df_novo['Data do Atestado'], dayfirst=True)


- Agora após tranformar o tipo das colunas vamos ver novamente como ficaram os tipos

In [9]:
df_novo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91 entries, 0 to 90
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   Código                91 non-null     int64         
 1   Custo do afastamento  89 non-null     object        
 2   Funcionário           91 non-null     object        
 3   Departamento          89 non-null     object        
 4   Data do Atestado      90 non-null     datetime64[ns]
 5   Especialidade         75 non-null     object        
 6   Motivo                85 non-null     object        
 7   Líder                 8 non-null      object        
dtypes: datetime64[ns](1), int64(1), object(6)
memory usage: 5.8+ KB


- Agora vamos preencher a coluna de líderes seguindo o exemplo do desafio, temos líderes e todos os funcionários abaixo dele são seus liderados, até chegar no proximo lider que lidera quem está abaixo dele

In [10]:
df_novo.head(15)

Código Custo do afastamento  Funcionário  \
0   1036743                 20,4  Anonimo 101   
1   1036742                   23    Anonimo 1   
2   1036741                 22,8    Anonimo 1   
3   1036740                  NaN    Anonimo 1   
4   1036739                   22    Anonimo 1   
5   1033172                  205    Anonimo 1   
6   1032228                  123    Anonimo 1   
7   1030902                  205    Anonimo 1   
8   1036621                  321    Anonimo 2   
9   1036749                 99.8  Anonimo 114   
10  1036762                  331    Anonimo 3   
11  1027641                  123    Anonimo 4   
12  1036760                92.36    Anonimo 5   
13  1037656                94.61  Anonimo 109   
14  1036733                94.61    Anonimo 9   

                       Departamento Data do Atestado          Especialidade  \
0                           Gerente       2019-05-29                 Exames   
1         ASSISTENTE DE IMPLANTACAO       2019-05-23  Neurologia pediátrica   
2         ASSISTENTE DE IMPLANTACAO       2019-05-17                    NaN   
3         ASSISTENTE DE IMPLANTACAO       2019-05-16                 Exames   
4         ASSISTENTE DE IMPLANTACAO       2019-05-06                    NaN   
5         ASSISTENTE DE IMPLANTACAO       2019-04-24              Pediatria   
6         ASSISTENTE DE IMPLANTACAO       2019-04-04  Neurologia pediátrica   
7         ASSISTENTE DE IMPLANTACAO       2019-03-07           Odontologia    
8                TECNICO SEGUROS IV       2019-05-21                    NaN   
9                           Gerente       2019-05-02                    NaN   
10               TECNICO SEGUROS II       2019-05-09                    NaN   
11  ANALISTA CONTROLE OPERACIONAL I       2019-03-13           Dermatologia   
12        ASSISTENTE OPERACIONAL II       2019-05-16             Neurologia   
13                          Gerente       2019-06-18              Nutrólogo   
14             TECNICO SEGUROS VG I       2019-05-27                    NaN   

                     Motivo Líder  
0                    Exames   Sim  
1   Acompanhamento familiar   NaN  
2           Consulta médica   NaN  
3                    Exames   NaN  
4           Consulta médica   NaN  
5   Acompanhamento familiar   NaN  
6   Acompanhamento familiar   NaN  
7     Consulta odontológica   NaN  
8                       NaN   NaN  
9           Consulta médica   Sim  
10                      NaN   NaN  
11          Consulta médica   NaN  
12          Consulta médica   NaN  
13          Consulta médica   Sim  
14                      NaN   NaN

- Ex: Anonimo 101 é lider de todos até o anonimo 114, que é lider de todos abaixo dele até o anonimo 109

- Usaremos um script python para fazer o preenchimento do nome dos lideres em cada coluna dos liderados, facilitando assim a identificação do lider de cada funcionário.

In [11]:

# Preencha os valores em branco na coluna 'Líder' com 'Não'
df_novo['Líder'].fillna('Não', inplace=True)

# Itere pelas linhas do DataFrame e preencha os líderes

lider_atual = 'Não'
for index, row in df_novo.iterrows():
    if row['Líder'] == 'Sim':
        lider_atual = row['Funcionário']
    else:
        df_novo.at[index, 'Líder'] = lider_atual

# Salve o DataFrame com os líderes preenchidos
df_novo.to_csv('funcionarios_com_lideres.csv', index=False)


* Esse código em Python usando a biblioteca Pandas realiza o preenchimento dos valores em branco na coluna 'Líder' e atribui um líder com base na coluna 'Funcionário'. Vou explicar o código em detalhes:

1. `df_novo['Líder'].fillna('Não', inplace=True)`: Nesta linha, os valores em branco (NaN) na coluna 'Líder' são preenchidos com a string 'Não'. O parâmetro `inplace=True` indica que essa operação será feita no próprio DataFrame 'df_novo', ou seja, os valores em branco serão substituídos diretamente no DataFrame.

2. Em seguida, o código entra em um loop que itera por todas as linhas do DataFrame usando a função `iterrows()`. Essa função permite acessar cada linha do DataFrame uma a uma.

3. Dentro do loop, verifica-se se o valor da coluna 'Líder' na linha atual é igual a 'Sim'. Se for, significa que esta linha indica um novo líder. Nesse caso, a variável `lider_atual` é atualizada com o valor da coluna 'Funcionário' dessa linha.

4. Se o valor da coluna 'Líder' não for 'Sim', isso indica que o líder atual ainda é o mesmo da linha anterior, então a coluna 'Líder' é preenchida com o valor de 'lider_atual'.

5. Após a iteração por todas as linhas, o DataFrame 'df_novo' agora contém a coluna 'Líder' preenchida de acordo com a lógica especificada.



Esse código é útil quando você deseja preencher uma coluna com base em uma lógica específica, como atribuir líderes com base na presença do valor 'Sim' na coluna 'Líder'. Certifique-se de que seus dados originais estejam corretamente carregados no DataFrame 'df_novo' antes de executar esse código.

- Vamos analisar novamente como ficou nosso dataframe.

In [12]:

df_novo.head()

Código Custo do afastamento  Funcionário               Departamento  \
0  1036743                 20,4  Anonimo 101                    Gerente   
1  1036742                   23    Anonimo 1  ASSISTENTE DE IMPLANTACAO   
2  1036741                 22,8    Anonimo 1  ASSISTENTE DE IMPLANTACAO   
3  1036740                  NaN    Anonimo 1  ASSISTENTE DE IMPLANTACAO   
4  1036739                   22    Anonimo 1  ASSISTENTE DE IMPLANTACAO   

  Data do Atestado          Especialidade                   Motivo  \
0       2019-05-29                 Exames                   Exames   
1       2019-05-23  Neurologia pediátrica  Acompanhamento familiar   
2       2019-05-17                    NaN          Consulta médica   
3       2019-05-16                 Exames                   Exames   
4       2019-05-06                    NaN          Consulta médica   

         Líder  
0          Sim  
1  Anonimo 101  
2  Anonimo 101  
3  Anonimo 101  
4  Anonimo 101

- Podemos ver que as linhas foram preenchidas corretamento com o nome de cada líder

- Vamos investigar se existem outliers no nosso dataframe, primeiro precisamos mudar a coluna "Custo do Afastamento" para float já que se trata de números e valores.

In [13]:
df_novo['Custo do afastamento'] = df_novo['Custo do afastamento'].str.replace(',', '.').astype(float)

- Agora vamos calcular os quartis da coluna 'Custo do afastamento' definindo limites inferiores e superiores.

In [14]:

# Calcule os quartis
q1 = df_novo['Custo do afastamento'].quantile(0.25)
q3 = df_novo['Custo do afastamento'].quantile(0.75)

# Calcule o IQR (intervalo interquartil)
iqr = q3 - q1

# Calcule os limites para identificar outliers
limite_inferior = q1 - 1.5 * iqr
limite_superior = q3 + 1.5 * iqr

# Encontre os outliers
outliers = df_novo[(df_novo['Custo do afastamento'] < limite_inferior) | (df_novo['Custo do afastamento'] > limite_superior)]

# Exiba somente os outliers
outliers


Código  Custo do afastamento  Funcionário  \
0   1036743                 20.40  Anonimo 101   
1   1036742                 23.00    Anonimo 1   
2   1036741                 22.80    Anonimo 1   
4   1036739                 22.00    Anonimo 1   
5   1033172                205.00    Anonimo 1   
7   1030902                205.00    Anonimo 1   
8   1036621                321.00    Anonimo 2   
10  1036762                331.00    Anonimo 3   
16  1036729                310.00  Anonimo 108   
17  1036730                310.00    Anonimo 8   
18  1033416                310.00    Anonimo 8   
19  1032282                310.00    Anonimo 8   
20  1032281                310.00    Anonimo 8   
22  1032230                310.00    Anonimo 8   
23  1032232                310.00    Anonimo 8   
24  1030893                310.00    Anonimo 8   
25  1027770                310.45    Anonimo 8   
26  1027769                310.00    Anonimo 8   
69  1027601               9215.00   Anonimo 15   

                           Departamento Data do Atestado  \
0                               Gerente       2019-05-29   
1             ASSISTENTE DE IMPLANTACAO       2019-05-23   
2             ASSISTENTE DE IMPLANTACAO       2019-05-17   
4             ASSISTENTE DE IMPLANTACAO       2019-05-06   
5             ASSISTENTE DE IMPLANTACAO       2019-04-24   
7             ASSISTENTE DE IMPLANTACAO       2019-03-07   
8                    TECNICO SEGUROS IV       2019-05-21   
10                   TECNICO SEGUROS II       2019-05-09   
16                              Gerente       2019-05-27   
17   ANALISTA INFORMACOES GERENCIAIS II       2019-05-23   
18                    TECNICO SEGUROS I       2019-04-22   
19   ANALISTA INFORMACOES GERENCIAIS II       2019-04-17   
20   ANALISTA INFORMACOES GERENCIAIS II       2019-04-16   
22   ANALISTA INFORMACOES GERENCIAIS II       2019-04-10   
23   ANALISTA INFORMACOES GERENCIAIS II       2019-04-03   
24   ANALISTA INFORMACOES GERENCIAIS II       2019-03-29   
25   ANALISTA INFORMACOES GERENCIAIS II       2019-03-21   
26   ANALISTA INFORMACOES GERENCIAIS II       2019-03-20   
69                 ANALISTA CONTABIL II       2019-03-07   

            Especialidade                   Motivo        Líder  
0                  Exames                   Exames          Sim  
1   Neurologia pediátrica  Acompanhamento familiar  Anonimo 101  
2                     NaN          Consulta médica  Anonimo 101  
4                     NaN          Consulta médica  Anonimo 101  
5               Pediatria  Acompanhamento familiar  Anonimo 101  
7            Odontologia     Consulta odontológica  Anonimo 101  
8                     NaN                      NaN  Anonimo 101  
10                    NaN                      NaN  Anonimo 114  
16          Cardiologista          Consulta médica          Sim  
17          Cardiologista                   Exames  Anonimo 108  
18          Cardiologista          Consulta médica  Anonimo 108  
19          Cardiologista                   Exames  Anonimo 108  
20          Cardiologista          Consulta médica  Anonimo 108  
22          Cardiologista                   Exames  Anonimo 108  
23           Fisioterapia                   Exames  Anonimo 108  
24              Ortopedia          Consulta médica  Anonimo 108  
25                 Outras                   Exames  Anonimo 108  
26              Ortopedia               Dor/Doença  Anonimo 108  
69                    NaN                   Exames  Anonimo 111

- Identificamos que temos vários valores fora dos limites estabelecidos, mas o que mais chama atenção é o valor da linha 69, pois o valor 310 aparece em várias ocorrências, já o 9215 só aparece uma unica vez

- Vamos procurar mais informações do funcionário "Anonimo 15"

In [15]:
func15 = df_novo[df_novo['Funcionário'] == 'Anonimo 15']

# Exiba o resultado
func15

Código  Custo do afastamento Funcionário           Departamento  \
58  1036620                 92.15  Anonimo 15   ANALISTA CONTABIL II   
59  1036752                 92.15  Anonimo 15   ANALISTA CONTABIL II   
60  1036781                 92.15  Anonimo 15   ANALISTA CONTABIL II   
61  1033423                 92.15  Anonimo 15   ANALISTA CONTABIL II   
62  1033425                 92.15  Anonimo 15   ANALISTA CONTABIL II   
63  1032179                 92.15  Anonimo 15   ANALISTA CONTABIL II   
64  1032180                 92.15  Anonimo 15   ANALISTA CONTABIL II   
65  1032181                 92.15  Anonimo 15   ANALISTA CONTABIL II   
66  1027607                 92.15  Anonimo 15   ANALISTA CONTABIL II   
67  1030882                 92.15  Anonimo 15   ANALISTA CONTABIL II   
68  1027604                 92.15  Anonimo 15   ANALISTA CONTABIL II   
69  1027601               9215.00  Anonimo 15   ANALISTA CONTABIL II   

   Data do Atestado            Especialidade           Motivo        Líder  
58       2019-05-22                      NaN              NaN  Anonimo 111  
59       2019-05-15  Ginecologia/Obstetricia           Exames  Anonimo 111  
60       2019-05-06                   Exames           Exames  Anonimo 111  
61       2019-04-30  Ginecologia/Obstetricia  Consulta médica  Anonimo 111  
62       2019-04-17                   Exames           Exames  Anonimo 111  
63       2019-04-10              Obstetricia       Dor/Doença  Anonimo 111  
64       2019-04-08              Obstetricia       Dor/Doença  Anonimo 111  
65       2019-04-01              Obstetricia  Exame periódico  Anonimo 111  
66       2019-03-21                      NaN           Exames  Anonimo 111  
67       2019-03-20                   Outras  Consulta médica  Anonimo 111  
68       2019-03-13                      NaN           Exames  Anonimo 111  
69       2019-03-07                      NaN           Exames  Anonimo 111

- Através dessa análise podemos observar que todos os valores do "Anonimo 15" foram de "92.15", o que nos leva a acreditar que o valor da linha 69 foi um erro de digitação por ter dois zeros a mais, vamos trocar o valor para que fique igual aos outros, 

In [16]:

# Acessando o valor da linha 69  na coluna 'Custo do afastamento' e dividindo por 100
df_novo.loc[69, 'Custo do afastamento'] = df_novo.loc[69, 'Custo do afastamento'] / 100


- Agora podemos ver que os valores estão normalizados

In [17]:
func15 = df_novo[df_novo['Funcionário'] == 'Anonimo 15']

# Exiba o resultado
func15

Código  Custo do afastamento Funcionário           Departamento  \
58  1036620                 92.15  Anonimo 15   ANALISTA CONTABIL II   
59  1036752                 92.15  Anonimo 15   ANALISTA CONTABIL II   
60  1036781                 92.15  Anonimo 15   ANALISTA CONTABIL II   
61  1033423                 92.15  Anonimo 15   ANALISTA CONTABIL II   
62  1033425                 92.15  Anonimo 15   ANALISTA CONTABIL II   
63  1032179                 92.15  Anonimo 15   ANALISTA CONTABIL II   
64  1032180                 92.15  Anonimo 15   ANALISTA CONTABIL II   
65  1032181                 92.15  Anonimo 15   ANALISTA CONTABIL II   
66  1027607                 92.15  Anonimo 15   ANALISTA CONTABIL II   
67  1030882                 92.15  Anonimo 15   ANALISTA CONTABIL II   
68  1027604                 92.15  Anonimo 15   ANALISTA CONTABIL II   
69  1027601                 92.15  Anonimo 15   ANALISTA CONTABIL II   

   Data do Atestado            Especialidade           Motivo        Líder  
58       2019-05-22                      NaN              NaN  Anonimo 111  
59       2019-05-15  Ginecologia/Obstetricia           Exames  Anonimo 111  
60       2019-05-06                   Exames           Exames  Anonimo 111  
61       2019-04-30  Ginecologia/Obstetricia  Consulta médica  Anonimo 111  
62       2019-04-17                   Exames           Exames  Anonimo 111  
63       2019-04-10              Obstetricia       Dor/Doença  Anonimo 111  
64       2019-04-08              Obstetricia       Dor/Doença  Anonimo 111  
65       2019-04-01              Obstetricia  Exame periódico  Anonimo 111  
66       2019-03-21                      NaN           Exames  Anonimo 111  
67       2019-03-20                   Outras  Consulta médica  Anonimo 111  
68       2019-03-13                      NaN           Exames  Anonimo 111  
69       2019-03-07                      NaN           Exames  Anonimo 111

- Vamos alterar os nomes da colunas, para não termos problemas de sintaxe no mysql

In [18]:

novos_nomes = {
    'Código': 'Codigo',
    'Custo do afastamento': 'Custo_do_afastamento',
    'Funcionário': 'Funcionario',
    'Departamento': 'Departamento',
    'Data do Atestado': 'Data_do_Atestado',
    'Especialidade': 'Especialidade',
    'Motivo': 'Motivo',
    'Líder': 'Lider'
}

# Renomeie as colunas do DataFrame com os novos nomes (sem acentos)
df_novo.rename(columns=novos_nomes, inplace=True)

- Por fim vamos salvar nossos dados tratados em um novo csv

In [19]:
df_novo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91 entries, 0 to 90
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   Codigo                91 non-null     int64         
 1   Custo_do_afastamento  89 non-null     float64       
 2   Funcionario           91 non-null     object        
 3   Departamento          89 non-null     object        
 4   Data_do_Atestado      90 non-null     datetime64[ns]
 5   Especialidade         75 non-null     object        
 6   Motivo                85 non-null     object        
 7   Lider                 91 non-null     object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(5)
memory usage: 5.8+ KB


In [20]:

df_novo.to_csv('Dados_Final.csv', index=False)

## Carregamentos dos Dados

- O arquivo Sql com o INSERT dos dados está com o nome de Dados_Final.sql


## Perguntas